src/prefect/context.py

```diff
@@ -210,6 +210,7 @@ def __enter__(self):
         self._context_stack += 1
         if self._context_stack == 1:
             self.client.__enter__()
+            self.client.api_compatible()
```

Comment:
This and below seemed like the best place to put this so that it only called once